### PR TITLE
Prow migration from 3.11 -> 4.x on build01

### PR DESCRIPTION
**Pull request by deepsm007**
Migrating the sshd-bastion to build01
Closed and Separated the process of migration #16218, #16237, #16228, #16242, #16246
Signed off: Deep Mistry

### CODE REVIEW

**Comment by hongkailiu** [comment]
I will review today.

File: core-services/sanitize-prow-jobs/_config.yaml [hunk 1]
```diff
@@ -3838,8 +3838,8 @@ groups:
     - periodic-ci-openshift-cluster-autoscaler-operator-e2e-azure-master
     - periodic-ci-openshift-cluster-autoscaler-operator-e2e-gcp-master
     - periodic-openshift-library-import
-    - release-openshift-origin-installer-e2e-remote-libvirt-ppc64le-4.7
-    - release-openshift-origin-installer-e2e-remote-libvirt-s390x-4.7
+    - release-openshift-ocp-installer-e2e-remote-libvirt-ppc64le-4.7
```
Comment (hongkailiu) [inline core-services/sanitize-prow-jobs/_config.yaml:3841]: This is the missing step in the card.
We can remove those job names in the config file.
They are controlled by `sshBastion: build01`, which assigns all the jobs with `dptp.openshift.io/ssh-bastion: "true"` label the `build01` regardless of what the rest of the config file looks like.

File: clusters/build-clusters/01_cluster/sshd-bastion/OWNERS [hunk 2]
```diff
@@ -0,0 +1,2 @@
+approvers:
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/OWNERS:1]: Are those file with the same contents as before?
Moving is fine, no need for copying.
It wont be deleted on `api.ci` after merging even in the case of moving the files over.
We have to manually delete the resources on `api.ci` when the jobs work fine on `build01`.

File: ci-operator/jobs/openshift/release/openshift-release-release-4.7-periodics.yaml [hunk 3]
```diff
@@ -526,7 +526,7 @@ periodics:
       secret:
         secretName: ci-pull-credentials
 - agent: kubernetes
-  cluster: api.ci
+  cluster: build02
```
Comment (hongkailiu) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.7-periodics.yaml:529]: This one should not be moved.

File: clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_config_updater_rbac.yaml [hunk 4]
```diff
@@ -5,7 +5,6 @@ items:
   apiVersion: rbac.authorization.k8s.io/v1
   metadata:
     name: config-updater
-    namespace: bastion-z
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_config_updater_rbac.yaml:8]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_config_updater_rbac.yaml [hunk 5]
```diff
@@ -30,7 +29,6 @@ items:
...
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_config_updater_rbac.yaml:33]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_owners_rbac.yaml [hunk 6]
```diff
@@ -5,7 +5,6 @@ items:
   apiVersion: rbac.authorization.k8s.io/v1beta1
   metadata:
     name: admin-role
-    namespace: bastion-ppc64le-libvirt
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_owners_rbac.yaml:8]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_rbac.yaml [hunk 7]
```diff
@@ -4,17 +4,15 @@ items:
 - kind: Namespace
   apiVersion: v1
   metadata:
-    name: bastion-telco
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_rbac.yaml:7]: Why is this changed?

File: clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_rbac.yaml [hunk 8]
```diff
@@ -4,17 +4,15 @@ items:
...
+    name: bastion-ppc64le-libvirt
 - kind: ServiceAccount
   apiVersion: v1
   metadata:
     name: port-forwarder
-    namespace: bastion-telco
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_rbac.yaml:12]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_rbac.yaml [hunk 9]
```diff
@@ -4,17 +4,15 @@ items:
...
 - kind: Role
   apiVersion: rbac.authorization.k8s.io/v1
   metadata:
     name: port-forward
-    namespace: bastion-telco
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/ppc64le-libvirt/admin_rbac.yaml:17]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/z/admin_config_updater_rbac.yaml [hunk 10]
```diff
@@ -5,7 +5,6 @@ items:
   apiVersion: rbac.authorization.k8s.io/v1
   metadata:
     name: config-updater
-    namespace: bastion-telco
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/z/admin_config_updater_rbac.yaml:8]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/z/admin_config_updater_rbac.yaml [hunk 11]
```diff
@@ -30,7 +29,6 @@ items:
...
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/z/admin_config_updater_rbac.yaml:33]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/z/admin_owners_rbac.yaml [hunk 12]
```diff
@@ -5,7 +5,6 @@ items:
   apiVersion: rbac.authorization.k8s.io/v1beta1
   metadata:
     name: admin-to-jaypoulz
-    namespace: bastion-z
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/z/admin_owners_rbac.yaml:8]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/z/admin_rbac.yaml [hunk 13]
```diff
@@ -9,12 +9,10 @@ items:
   apiVersion: v1
   metadata:
     name: port-forwarder
-    namespace: bastion-z
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/z/admin_rbac.yaml:12]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/z/admin_rbac.yaml [hunk 14]
```diff
@@ -9,12 +9,10 @@ items:
...
 - kind: Role
   apiVersion: rbac.authorization.k8s.io/v1
   metadata:
     name: port-forward
-    namespace: bastion-z
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/z/admin_rbac.yaml:17]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/z/admin_rbac.yaml [hunk 15]
```diff
@@ -33,13 +31,10 @@ items:
   apiVersion: rbac.authorization.k8s.io/v1beta1
   metadata:
     name: port-forward-to-port-forwarder
-    namespace: bastion-z
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/z/admin_rbac.yaml:36]: This stays.

File: clusters/build-clusters/01_cluster/sshd-bastion/z/admin_rbac.yaml [hunk 16]
```diff
@@ -33,13 +31,10 @@ items:
...
   subjects:
   - kind: ServiceAccount
     name: port-forwarder
-    namespace: bastion-z
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/sshd-bastion/z/admin_rbac.yaml:40]: This stays.

File: core-services/sanitize-prow-jobs/_config.yaml [hunk 17]
```diff
@@ -783,7 +783,6 @@ buildFarm:
       - .*openshift-origin-branding-release-4.4-presubmits.yaml$
       - .*openshift-origin-web-console-enterprise-3.11-presubmits.yaml$
       - .*openshift-origin-web-console-enterprise-3.9-presubmits.yaml$
-      - .*openshift-os-master-presubmits.yaml$
```
Comment (hongkailiu) [inline core-services/sanitize-prow-jobs/_config.yaml:786]: Why is this removed?

**Comment by hongkailiu** [comment]
The general rule is to remove the namespace field under `roleref`.

**Comment by deepsm007** [comment]
Separated the process of migration https://github.com/openshift/release/pull/16218, https://github.com/openshift/release/pull/16237, https://github.com/openshift/release/pull/16228, https://github.com/openshift/release/pull/16242, https://github.com/openshift/release/pull/16246
